### PR TITLE
[Cleanup] Delete Linalg on buffers pipeline

### PR DESCRIPTION
**Pull request by hanhanW**
There are four commits in the PR:

1. Delete clEnableLinalgOnTensorsDispatch option in Flow dialect
2. Make SPIR-V usingLinalgOnTensors true by default
3. Delete Linalg on tensors/buffers option for LLVM backend
4. Delete Linalg on buffers pass

The option for Linalg on tensors/buffers for SPIR-V path is not deleted because some lit tests relied on it.

note: this also cleans up some deps.

### CODE REVIEW

**Comment by hanhanW** [comment]
I'd suggest to review through three commits individually.

File: iree/compiler/Conversion/LinalgToSPIRV/Passes.cpp [hunk 1]
```diff
@@ -209,18 +209,6 @@ void buildSPIRVTransformPassPipeline(OpPassManager &pm,
         Shape::createMaterializeShapeCalculationsPass());
     pm.nest<ModuleOp>().addNestedPass<FuncOp>(
         Shape::createHoistShapeCalculationsPass());
-
```
Comment (antiagainst) [inline iree/compiler/Conversion/LinalgToSPIRV/Passes.cpp:212]: Why we keep the shape related stuff untouched in the `else` branch? Can't we delete the `else` branch entirely?

Comment (hanhanW) [reply iree/compiler/Conversion/LinalgToSPIRV/Passes.cpp:212]: Let me take a stab at it. I thought some un-cleanup lit tests rely on these things. I only verified that there are no mhlo ops in lit tests, so it is totally safe to delete the pipeline.

Comment (hanhanW) [reply iree/compiler/Conversion/LinalgToSPIRV/Passes.cpp:212]: ah, it also works.

**Review by antiagainst — APPROVED** [review]
Hooray! LGTM. But please wait for @MaheshRavishankar's approval.

File: iree/compiler/Conversion/HLOToLinalg/BUILD [hunk 2]
```diff
@@ -51,35 +51,21 @@ cc_library(
 cc_library(
     name = "HLOToLinalg",
     srcs = [
-        "HLOToLinalgOnBuffers.cpp",
-        "Passes.cpp",
         "ResolveShapeOps.cpp",
```
Comment (MaheshRavishankar) [inline iree/compiler/Conversion/HLOToLinalg/BUILD:54]: Interesting! I think `ResolveShapeOps.cpp` can go too, but it is used for now, but probably not needed. Not needed to be resolved here, just a note for (my) future reference.

File: iree/compiler/Dialect/Flow/Transforms/Passes.cpp [hunk 3]
```diff
@@ -74,11 +68,9 @@ namespace Flow {
 static void buildHLOInputTransformPassPipeline(OpPassManager &passManager) {
   passManager.addNestedPass<FuncOp>(
       IREE::Flow::createHLOToHLOPreprocessingPass());
-  if (clEnableLinalgOnTensorsDispatch) {
-    // TODO(ataei): This should run as part of createHLOToHLOPreprocessingPass
-    // which will break VMLA backend.
-    passManager.addNestedPass<FuncOp>(createDecomposeHLOClampPass());
-  }
+  // TODO(ataei): This should run as part of createHLOToHLOPreprocessingPass
```
Comment (MaheshRavishankar) [inline iree/compiler/Dialect/Flow/Transforms/Passes.cpp:71]: Does this not break VMLA backend?

**Review by MaheshRavishankar — APPROVED** [review]
Great! That cleans up a lot of stuff. SOme of the individual passes can be removed as well. Will do so.

**Review by ThomasRaoux — APPROVED** [review]

File: iree/compiler/Dialect/Flow/Transforms/Passes.cpp (hunk 3 above)
Comment (asaadaldien) [reply iree/compiler/Dialect/Flow/Transforms/Passes.cpp:71]: It use to, not all HLO ops here has VMLA equivalent

**Review by asaadaldien — APPROVED** [review]
Yay....HaPpYDaNcE!

File: iree/compiler/Dialect/Flow/Transforms/Passes.cpp (hunk 3 above)
Comment (hanhanW) [reply iree/compiler/Dialect/Flow/Transforms/Passes.cpp:71]: No it does not.

File: iree/compiler/Conversion/HLOToLinalg/BUILD (hunk 2 above)
Comment (hanhanW) [reply iree/compiler/Conversion/HLOToLinalg/BUILD:54]: I tried to remove it, but it's used for old lit tests. I plan to delete it later.
And it also broke regressiondynamic_add.mlir, but I did not dig into it.

**Pull request merged**